packages/types/src/transaction/view-transaction-plan/view-action-plan.ts

```diff
@@ -0,0 +1,154 @@
+import {
+  ActionPlan,
+  ActionView,
+} from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/transaction/v1alpha1/transaction_pb';
+import {
+  DenomMetadata,
+  Value,
+  ValueView,
+} from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/asset/v1alpha1/asset_pb';
+import {
+  Note,
+  NoteView,
+  OutputPlan,
+  OutputView,
+  SpendPlan,
+  SpendView,
+} from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/component/shielded_pool/v1alpha1/shielded_pool_pb';
+import { Jsonified } from '../../jsonified';
+import { bech32AssetId } from '../../asset';
+
+const getValueView = (
+  value: Value | undefined,
+  denomMetadataByAssetId: Record<string, Jsonified<DenomMetadata>>,
+): ValueView => {
+  if (!value) throw new Error('No value to view');
+  if (!value.assetId) throw new Error('No asset ID in value');
+  if (!value.amount) throw new Error('No amount in value');
+
+  const denomMetadata = denomMetadataByAssetId[bech32AssetId(value.assetId)];
+  if (!denomMetadata) throw new Error('Asset ID refers to an unknown asset type');
+
+  return new ValueView({
+    valueView: {
+      case: 'knownDenom',
+      value: {
+        amount: value.amount,
+        denom: DenomMetadata.fromJson(denomMetadata),
+      },
+    },
+  });
+};
+
+const getNoteView = (
+  note: Note | undefined,
+  denomMetadataByAssetId: Record<string, Jsonified<DenomMetadata>>,
+) => {
+  if (!note) throw new Error('No note to view');
+  if (!note.address) throw new Error('No address in note');
+  if (!note.value) throw new Error('No value in note');
+
+  return new NoteView({
+    address: {
+      addressView: {
+        case: 'opaque',
+        value: { address: note.address },
+      },
+    },
+    value: getValueView(note.value, denomMetadataByAssetId),
+  });
+};
+
+const getSpendView = (
+  spendPlan: SpendPlan,
+  denomMetadataByAssetId: Record<string, Jsonified<DenomMetadata>>,
+): SpendView => {
+  if (!spendPlan.note?.address) throw new Error('No address in spend plan');
+
+  return new SpendView({
+    spendView: {
+      case: 'visible',
+      value: {
+        note: getNoteView(spendPlan.note, denomMetadataByAssetId),
+      },
+    },
+  });
+};
+
+const getOutputView = (
+  outputPlan: OutputPlan,
+  denomMetadataByAssetId: Record<string, Jsonified<DenomMetadata>>,
+): OutputView => {
+  if (!outputPlan.destAddress) throw new Error('No destAddress in output plan');
+
+  return new OutputView({
```

Comment:
Is there any add'l data we need in this `OutputView`? Didn't look like it to me, but want to make sure I'm not missing something.